schema/ksl-schema.kdl

```diff
@@ -0,0 +1,925 @@
+                }
+                prop lang about="Link language\n\nLanguage of the destination document." {
+                    ref metadata-lang
+                }
+            }
+            node license about="""
+                Schema license
+
+                The license(s) that the schema is licensed under.
+                """ {
+                repeatable
+                arg description="Name of the used license" {
+                    type string
+                }
+                prop spdx description="An SPDX license identifier" {
+                    type string
+                    // TODO: validation?
+                }
+                prop path about="Path to a local license file" {
+                    type string
+                }
+                prop url about="URL to an externally-stored license" {
+                    type string
+                    format url url-reference irl irl-reference
+                }
+                children {
+                    node link {
+                        ref metadata-link
+                    }
+                }
+            }
+            node published about="""
+                Schema publication date
+
+                Date or data+time when the schema was published.
+                """ {
+                arg about="Publication date" {
+                    type string
+                    format date date-time
+                }
+            }
+            node modified about="""
+                Schema modification date
+
+                When the schema was modified. If used multiple times, the most
+                recent date will be considered 'latest'.
+                """ {
+                repeatable
+                args about="Modification date" {
+                    type string
+                    format date date-time
+                }
+            }
+            node version about="""
+                Schema semver version
+
+                The version number of this version of the schema, in semver
+                format.
+                """ {
+                arg about="Semver version number" {
+                    type string
+                    // https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string.
+                    pattern #"^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"#
+                }
+            }
+        }
+    }
+    node definitions about="""
+        Inert validation definitions
+
+        An optional set of definitions that may be referenced elsewhere in the
+        schema. They will be inert (that is, not directly apply to the document)
+        unless referenced by another node inside `document`.
+        """ {
+        children {
+            allow-others
+        }
+    }
+    node document {
+        ref children-node
+        default {
```

Comment:
As I mentioned in my (too long) reply about this, I think I might prefer to not have the concept of `default` nodes... Though perhaps that's just because I'm struggling to see a use for them outside of making schema-schemas?

Reply:
I'll remove default nodes. At least for now.